dbms/src/Storages/DeltaMerge/Segment.cpp

```diff
@@ -316,7 +316,10 @@ bool Segment::write(DMContext & dm_context, const Block & block)
 
     if (delta->appendColumnFile(dm_context, column_file))
     {
-        flushCache(dm_context);
+        if (flush_cache)
+        {
+            flushCache(dm_context);
```

Comment:
Does it need to check the flushCache result?

Reply:
This is only used in tests. So the flushCache cannot fail here. And even the flushCache fail can help us cover more scenario.

Reply:
Hmmm, "even the flushCache fail can help us cover more scenario" sorry I don't get the sense here. Can you explain more?
Think that even if methods are only used for tests, we also need to guarantee correctness.

Reply:
Got it and have fixed it.